### PR TITLE
HOSTEDCP-1412 | Extract hypershift version validation to a separate function

### PR DESCRIPTION
**Pull request by muraee**
- Extract hypershift version validation to a separate function `ValidateHypershiftVersion()`
- Update `LowestHostedCpSupport` to 4.14.0

### CODE REVIEW

**Comment by den-rgb** [comment]
Please include a test for the new function for the codecov to pass

**Comment by muraee** [comment]
@den-rgb shouldn't it be covered by https://github.com/openshift/rosa/blob/master/cmd/create/cluster/cmd_test.go#L148-L159?

**Comment by den-rgb** [comment]
@muraee I think it's giving out because it is function specific, and the tests cover `ValidateVersion` as a whole, I think adding a couple of tests that test the list response should fix it.

File: pkg/helper/versions/helpers_test.go [hunk 1]
```diff
@@ -65,16 +65,16 @@ var _ = Describe("Version Helpers", Ordered, func() {
...
-				"4.14.0-0.nightly-2023-02-27-084419",
-				"4.12.0",
+				"4.16.0-0.nightly-2023-02-27-084419",
+				"4.14.0",
 			),
 			Entry("Current control plane",
-				"4.12.5",
-				"4.12.0-0.a",
+				"4.14.5",
+				"4.14.0",
```
Comment (stevekuznetsov) [inline pkg/helper/versions/helpers_test.go:77]: Side-comment for the maintainers, this seems like a source of infinite toil as the minimum supported version keeps creeping up in the service. Perhaps possible to factor this some other way so the tests remain valid but the work is not necessary on every bump.

**Comment by muraee** [comment]
@den-rgb done

**Comment by muraee** [comment]
@den-rgb @robpblake could you approve if it looks good to you, thanks!

**Pull request merged**